src/Symfony/Component/Routing/RouteCollection.php

```diff
@@ -307,4 +348,37 @@ public function addResource(ResourceInterface $resource)
             $this->resources[$key] = $resource;
         }
     }
+
+    /**
+     * Sets an alias for an existing route.
+     *
+     * @param string $name  The alias to create
+     * @param string $alias The route to alias
+     *
+     * @throws InvalidArgumentException if the id is not a string or an Alias
+     * @throws InvalidArgumentException if the alias is for itself
+     */
+    public function addAlias(string $name, string $alias): Alias
+    {
+        if ($name === $alias) {
+            throw new InvalidArgumentException(sprintf('An alias can not reference itself, got a circular reference on "%s".', $name));
+        }
+
+        unset($this->routes[$name], $this->priorities[$name]);
+
+        return $this->aliases[$name] = new Alias($alias);
+    }
+
+    /**
+     * @return array<string, Alias>
+     */
+    public function getAliases(): array
+    {
+        return $this->aliases;
+    }
+
+    public function getAlias(string $name): ?Alias
```

Comment:
I removed `hasAlias()` and made `getAlias()` nullable instead, for consistency with `get()`